### ISSUE TITLE
[Bug]: AndroidJavaException: java.lang.ClassNotFoundException: com.onesignal.OneSignal

### ISSUE THREAD

**Issue by GorkemSvn** [OneSignal/OneSignal-Unity-SDK#562]
### What happened?

Update: I've figured out unity's proguard minify doesn't include some dependencies onesignal needs, resulting one signal initialization to fail. I can't disable minify because i can't build without it. So onesignal needs to patch itself to be compatible with proguard minify.

It looks like unity's single dex file is not enought to hold firebase firestore,auth and onesignal and using minify breaks things

I'm followed one signal unity sdk setup steps , no problems in editor. Wrote single line of code in my notificationsmanager c# script( OneSignal.Default.Initialize("xcvsdv");) But when i had built in my samsung phone a70, i didn't receive any notification ( i sent notification from dashbouard.) I noticed my phone haven't subscribed to audience. And got this message from autoconnected profiler

note: i did not do steps for amazon-store

AndroidJavaException: java.lang.ClassNotFoundException: com.onesignal.OneSignal
![image](https://user-images.githubusercontent.com/43960967/203917182-1ad33884-2d82-4231-ae07-961ebf4cb1da.png)

![image](https://user-images.githubusercontent.com/43960967/203917477-cd1034fd-3cdc-4ee9-9e68-1c20ac48fb4d.png)

### Steps to reproduce?

```Markdown
1. open empty project
2. import onesignal sdk
3. use unity's minify at publish settings to build
4. enable autoconnect profiler
5. classnotfound exception: com.onesignal.onesignal at mobile phone
```

### What did you expect to happen?

App to subscribe to onesignal audience

### Unity version

2020.3.42 

### OneSignal Unity SDK version

3.0.6

### Platform

Android

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by GorkemSvn** [comment]
I've been able to get some part of stacktrace:

![image](https://user-images.githubusercontent.com/43960967/203921298-e2c2c305-5c61-4f41-aefd-7a364eee00f1.png)

**Comment by GorkemSvn** [comment]
this is what happens when you install firebase firestore,auth and onesignal and does not minify. 

D8: Cannot fit requested classes in a single dex file (# methods: 114793 > 65536)
com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
The number of method references in a .dex file cannot exceed 64K.
Learn how to resolve this issue at https://developer.android.com/tools/building/multidex.html
	at com.android.builder.dexing.D8DexArchiveMerger.getExceptionToRethrow(D8DexArchiveMerger.java:132)
	at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:119)
	at com.android.build.gradle.internal.transforms.DexMergerTransformCallable.call(DexMergerTransformCallable.java:102)
	at com.android.build.gradle.internal.tasks.DexMergingTaskRunnable.run(DexMergingTask.kt:441)
	at com.android.build.gradle.internal.tasks.Workers$ActionFacade.run(Workers.kt:242)
	at org.gradle.workers.internal.AdapterWorkAction.execute(AdapterWorkAction.java:50)
	at org.gradle.workers.internal.DefaultWorkerServer.execute(DefaultWorkerServer.java:50)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:63)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1$1.create(NoIsolationWorkerFactory.java:59)
	at org.gradle.internal.classloader.ClassLoaderUtils.executeInClassloader(ClassLoaderUtils.java:98)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1.lambda$execute$0(NoIsolationWorkerFactory.java:59)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:44)
	at org.gradle.workers.internal.AbstractWorker$1.call(AbstractWorker.java:41)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:416)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$CallableBuildOperationWorker.execute(DefaultBuildOperationExecutor.java:406)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor$1.execute(DefaultBuildOperationExecutor.java:165)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:250)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.execute(DefaultBuildOperationExecutor.java:158)
	at org.gradle.internal.operations.DefaultBuildOperationExecutor.call(DefaultBuildOperationExecutor.java:102)
	at org.gradle.internal.operations.DelegatingBuildOperationExecutor.call(DelegatingBuildOperationExecutor.java:36)
	at org.gradle.workers.internal.AbstractWorker.executeWrappedInBuildOperation(AbstractWorker.java:41)
	at org.gradle.workers.internal.NoIsolationWorkerFactory$1.execute(NoIsolationWorkerFactory.java:53)
	at org.gradle.workers.internal.DefaultWorkerExecutor.lambda$submitWork$2(DefaultWorkerExecutor.java:200)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runExecution(DefaultConditionalExecutionQueue.java:215)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.runBatch(DefaultConditionalExecutionQueue.java:164)
	at org.gradle.internal.work.DefaultConditionalExecutionQueue$ExecutionRunner.run(DefaultConditionalExecutionQueue.java:131)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at org.gradle.internal.concurrent.ExecutorPolicy$CatchAndRecordFailures.onExecute(ExecutorPolicy.java:64)
	at org.gradle.internal.concurrent.ManagedExecutorImpl$1.run(ManagedExecutorImpl.java:48)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at org.gradle.internal.concurrent.ThreadFactoryImpl$ManagedThreadRunnable.run(ThreadFactoryImpl.java:56)
	at java.lang.Thread.run(Thread.java:748)
Caused by: com.android.tools.r8.CompilationFailedException: Compilation failed to complete
	at com.android.tools.r8.utils.W.a(:87)
	at com.android.tools.r8.D8.run(:11)
	at com.android.builder.dexing.D8DexArchiveMerger.mergeDexArchives(D8DexArchiveMerger.java:117)
	... 33 more
Caused by: com.android.tools.r8.utils.b: Error: null, Cannot fit requested classes in a single dex file (# methods: 114793 > 65536)
	at Version.fakeStackEntry(Version_2.0.88.java:0)
	at com.android.tools.r8.utils.O0.a(:21)
	at com.android.tools.r8.dex.P.a(:56)
	at com.android.tools.r8.dex.P$h.a(:7)
	at com.android.tools.r8.dex.b.a(:15)
	at com.android.tools.r8.dex.b.b(:24)
	at com.android.tools.r8.D8.d(:92)
	at com.android.tools.r8.D8.b(:1)
	at com.android.tools.r8.utils.W.a(:30)
	... 35 more


FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':launcher:mergeDexDebug'.
> A failure occurred while executing com.android.build.gradle.internal.tasks.Workers$ActionFacade
   > com.android.builder.dexing.DexArchiveMergerException: Error while merging dex archives: 
     The number of method references in a .dex file cannot exceed 64K.
     Learn how to resolve this issue at https://developer.android.com/tools/building/multidex.html

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 4s
Picked up JAVA_TOOL_OPTIONS: -Dfile.encoding=UTF-8

UnityEngine.GUIUtility:ProcessEvent (int,intptr,bool&)

**Comment by GorkemSvn** [comment]
Update
 I've updated sdk to 3.0.7
Created an empty project(unity 2020.3.42), imported onesignal sdk. get a build while minify is off. Saw my phone succesfully register to audience of onesignal
enabled the minify, cleared the audience from onesignal site. got another build
no audience registered. fail

**Comment by ridwanfauzy** [comment]
> Perbarui Saya telah memperbarui sdk ke 3.0.7 Membuat proyek kosong (unity 2020.3.42), mengimpor sdk satu sinyal. dapatkan build saat minify tidak aktif. Melihat ponsel saya berhasil mendaftar ke audiens onesignal mengaktifkan minify, menghapus audiens dari situs onesignal. mendapat build lain, tidak ada audiens yang terdaftar. gagal

can you use minify now?

**Comment by jkasten2** [comment]
Can you try adding this line to your Proguard config:
```
-keep com.onesignal.** { *; }
```
This will ensure all OneSignal classes are kept when minifying. You can consider this as a work around for now.

We will look into designing a Proguard config that can still allow some minification to OneSignal and include it in a future update so a manual step won't alway be required.

**Comment by GorkemSvn** [comment]
Okay I’ll try it. Thanks!

**Comment by pus2meong** [comment]
I have the same issue.
I test on editor, no error found.
I build the apk, and test it on my android phone, nothing happen. It should have asking to subscribe or something, right?
I check on onesignal dashboard, no entry at all.

I found out that the code went error when calling this line: OneSignal.Initialize("MY ONE SIGNAL ID");
Using try and catch in the build apk, I managed to print to text UI, the error message from exception variable: java.lang.ClassNotFoundException: com.onesignal.OneSignal

I don't use proguard, I just follow the step from onesignal unity sdk setup steps.
I don't use any other plugins, just this one.

**Comment by jacoposcala** [comment]
Same issue.
No proguard/minify active.
Using the last OneSignal version "5.0.5" and Unity 2022.3.2.
Getting_ java.lang.ClassNotFoundException: com.onesignal.OneSignal on android build

**Comment by jkasten2** [comment]
On this `ClassNotFoundException` with OneSignal-Unity-SDK v5 versions, this should have been fixed when updating to [OneSignal-Unity-SDK 5.0.3](https://github.com/OneSignal/OneSignal-Unity-SDK/releases/tag/5.0.3), however it is important you follow this step from the release notes if coming from an older SDK version:
> You must run the "Copy Android plugin to Assets" step in Window > OneSignal SDK Setup.

Please let us know if following this step fixes the issue for you. If not please provide an example Unity project so we can produce the issue on our end.

**Comment by shepherd-l** [comment]
Closing due to inactivity. If you are still experiencing this issue, please open a new issue with updated information
